src/tests/unit_tests/test_pred_prob.py

```diff
@@ -157,3 +157,79 @@ def test_simple_predictive_probability_unobserved(seed=0):
     Y = [(N_ROWS, 2, 4), (N_ROWS+1, 2, 5)]
     with pytest.raises(IndexError):
         vals = engine.simple_predictive_probability(M_c, X_L, X_D, Y, Q[1])
+
+def test_predictive_probability_observed(seed=0):
+    # TODO
+    pass
+
+def test_predictive_probability_unobserved(seed=0):
+    # This function tests the predictive probability for the joint distirbution.
+    # Throughout, we will check that the result is the same for the joint and
+    # simple calls.
+    T, M_r, M_c, X_L, X_D, engine = quick_le(seed)
+
+    # Hypothetical column number should throw an error.
+    Q = [(N_ROWS, 1, 1.5), (N_ROWS, 10, 2)]
+    Y = []
+    with pytest.raises(ValueError):
+        vals = engine.predictive_probability(M_c, X_L, X_D, Y, Q)
```

Comment:
Why are all these variables named `vals`, when the predictive probability should be a single number?  We should change them to singular and assert a few times that the result has the expected type.  Yes, that means some unpacking of the comparative `simple_predictive_probability` may be necessary.